bool flag
bool found_line
bool complete_circle
float32 obstacle_size
float64 distance_to_obstacle
float64[] initial_position
float64[] obstacle